models/mart/current_mrr_by_items.sql

```diff
@@ -23,42 +23,70 @@ with mrr_by_items as (
-)
-
---setting the updated date for the initial run
-{% if not is_incremental() %}
-    select 
-    customer_id,
-    customer_name,
-    subscription_item_id,
-	product_id,
-    price_id,
-    mrr,
-    brl_mrr,
-    stripe_account,
-    updated_at
-from mrr_by_items
-{% endif %} 
+),
 
---update date when MRR changes or when MRR is new 
-{% if is_incremental() %}
-select * from
-    (select
+updated_or_new_items as (
+    select
         mrr_by_items.customer_id,
         mrr_by_items.customer_name,
         mrr_by_items.subscription_item_id,
         mrr_by_items.product_id,
         mrr_by_items.price_id,
         mrr_by_items.mrr,
         mrr_by_items.brl_mrr,
-        mrr_by_items.stripe_account
-        ,case 
-            when current_mrr.updated_at is null then mrr_by_items.updated_at
-            when mrr_by_items.mrr <> current_mrr.mrr then mrr_by_items.updated_at
-            else current_mrr.updated_at
-        end as updated_at
+        mrr_by_items.stripe_account,
+        mrr_by_items.updated_at
+    from mrr_by_items
+    left join {{this}} as current_mrr on current_mrr.subscription_item_id = mrr_by_items.subscription_item_id
+    where mrr_by_items.mrr <> coalesce(current_mrr.mrr, 0) or current_mrr.subscription_item_id is null
+),
+
+-- Consulta para identificar itens removidos e manter suas informações, exceto MRR atualizado para zero
+removed_items as (
+    select
+        current_mrr.customer_id,
+        current_mrr.customer_name,
+        current_mrr.subscription_item_id,
+        current_mrr.product_id,
+        current_mrr.price_id,
+        0 as mrr, -- Atualiza MRR para zero
+        0 as brl_mrr, -- Assumindo que você queira zerar isso também
+        current_mrr.stripe_account,
+        coalesce(updated_at, now()) as updated_at -- Data da atualização para zero
+    from {{this}} as current_mrr
+    where not exists (
+        select 1
+        from mrr_by_items
+        where mrr_by_items.subscription_item_id = current_mrr.subscription_item_id
+    )
+)
+
+{% if not is_incremental() %}
+    -- Sua lógica original para o carregamento inicial
+    select 
+        customer_id,
+        customer_name,
+        subscription_item_id,
+        product_id,
+        price_id,
+        mrr,
+        brl_mrr,
+        stripe_account,
+        updated_at
     from mrr_by_items
-        left join {{this}} as current_mrr on current_mrr.subscription_item_id = mrr_by_items.subscription_item_id
-        where mrr_by_items.mrr <> current_mrr.mrr OR current_mrr.mrr is null
-        ) as incremental_update
-{% endif %} 
+{% else %}
+    -- União das consultas para itens atualizados ou novos, e itens removidos mantendo as informações existentes
+    select * from updated_or_new_items
+    union all
+    select
+        customer_id,
+        customer_name,
+        subscription_item_id,
+        product_id,
+        price_id,
+        mrr,
+        brl_mrr,
+        stripe_account,
+        updated_at
```

Comment:
I think the column names are the same, maybe we can change to

```suggestion
        *
```